filer/models/filemodels.py

```diff
@@ -91,10 +99,17 @@ class File(PolymorphicModel, mixins.IconsMixin):
                     'file. File will be publicly accessible '
                     'to anyone.'))
 
+    mime_type = models.CharField(
+        max_length=255,
+        help_text='MIME type of uploaded content',
+        validators=[mimetype_validator],
+        default='application/octet-stream',
+    )
+
     objects = FileManager()
 
     @classmethod
-    def matches_file_type(cls, iname, ifile, request):
+    def matches_file_type(cls, iname, ifile, mime_type):
```

Comment:
This is part of the public and documented API and we need to advertise that people extending the base class (https://django-filer.readthedocs.io/en/latest/extending_filer.html#the-very-basics) will have to update the method in the subclasses

Reply:
do we want to remove `request`? I'd keep it

Reply:
I don't see any advantage keeping it. It is never used and I can hardly see any possible future use-case, where the `request` objects could be needed. So in my opinion it's better to keep the signature slim.